### PR TITLE
Merge run data into report

### PR DESCRIPTION
**Pull request by wottop**

### CODE REVIEW

**Review by tlhmerry9 — APPROVED** [review]
This makes sense to me, am moving back to needs review so another set of eyes will also approve.

**Comment by jirihnidek** [comment]
It is not related to this PR, but I can see following warning on stdout, when I run virt-who:

```
[root@thinkpad-t580 virt-who]# ./virt-who -s -c /root/virt-who.conf.d/local_libvirt.conf
/home/jhnidek/githup/candlepin/virt-who/virtwho/manager/subscriptionmanager/subscriptionmanager.py:141: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if key is 'ssl_port':
+-------------------------------------------+
           Configuration Status
+-------------------------------------------+
Configuration Name: local_libvirt
Source Status: success
Destination Status: failure
```

**Comment by jirihnidek** [comment]
Results of testing:

When I run `virt-who --status` and virt-who.service is not running, then I get result quickly:

```
[root@thinkpad-t580 virt-who]# time ./virt-who -s -j -c /root/virt-who.conf.d/local_libvirt.conf 
{
    "configurations": [
        {
            "name": "local_libvirt",
            "source": {
                "connection": "qemu:///system?no_tty=1",
                "status": "success",
                "last_successful_retrieve": null,
                "hypervisors": null,
                "guests": null
            },
            "destination": {
                "connection": "svice",
                "status": "success",
                "last_successful_send": null,
                "last_successful_send_job_status": null
            }
        }
    ]
}

real	0m2,279s
user	0m0,248s
sys	0m0,039s
```

When `virt-who.service` is running and at least guest-to-host mapping have been gathered and sent to candlepin server, then I wait for result of status too long as you can see on the following output:

```
[root@thinkpad-t580 virt-who]# time ./virt-who -s -j -c /root/virt-who.conf.d/local_libvirt.conf 
{
    "configurations": [
        {
            "name": "local_libvirt",
            "source": {
                "connection": "qemu:///system?no_tty=1",
                "status": "success",
                "last_successful_retrieve": "2021-07-14 14:50:47 UTC",
                "hypervisors": 1,
                "guests": 7
            },
            "destination": {
                "connection": "svice",
                "status": "success",
                "last_successful_send": "2021-07-14 14:50:48 UTC",
                "last_successful_send_job_status": "FINISHED"
            }
        }
    ]
}

real	0m17,313s
user	0m0,281s
sys	0m0,041s
```

Result is the same, when `-j` is not used:

```
[root@thinkpad-t580 virt-who]# time ./virt-who -s -c /root/virt-who.conf.d/local_libvirt.conf 
+-------------------------------------------+
           Configuration Status
+-------------------------------------------+
Configuration Name: local_libvirt
Source Status: success
Destination Status: success



real	0m17,281s
user	0m0,256s
sys	0m0,039s
```

I think that 17 seconds is too much. I guess that there have to be some locking issue. I will do further investigation.

File: tests/test_subscriptionmanager.py
Comment (jirihnidek) [inline tests/test_subscriptionmanager.py]: Why do you use this method here? Wouldn't be enough to modify mocked data accordingly? Something like:

```
...
'updated': [ { 'uuid': '123'} ]
...
```

File: tests/test_subscriptionmanager.py
Comment (jirihnidek) [inline tests/test_subscriptionmanager.py]: I suspect that this commented code was left here by coincident, right?

File: virtwho/virt/virt.py
Comment (jirihnidek) [inline virtwho/virt/virt.py]: When `job_id` is property, then more pythonic way of implementing setter would be following:

```python
@job_id.setter:
def job_id(self, id):
    self._job_id = id
```

File: virtwho/virt/virt.py
Comment (jirihnidek) [inline virtwho/virt/virt.py]: Maybe we could implement other setters in more pythonic way. Thoughts?

**Review by jirihnidek — CHANGES_REQUESTED** [review]
Thanks for the PR. I can confirm that the run data are included in report, when `--json` option is used, but it is necessary to wait for response too long. I have also some suggestions to code and I left comments in code.

**Comment by jirihnidek** [comment]
When I'm looking to `rhsm.log` or I use `-d` option, then I can see 15 second pause:

```
[root@thinkpad-t580 virt-who]# time ./virt-who -s -j -d -c /root/virt-who.conf.d/local_libvirt.conf 
2021-07-14 17:22:33,959 [virtwho.rhsm_log INFO] MainProcess(61149):MainThread @config.py:init_config:1564 - Using configuration passed in by -c/--configs; ignoring configuration files in '/etc/virt-who.d/'
2021-07-14 17:22:33,960 [virtwho.rhsm_log DEBUG] MainProcess(61149):MainThread @config.py:init_config:1603 - [global]: Value for "oneshot" not set, using default: False
2021-07-14 17:22:33,960 [virtwho.rhsm_log DEBUG] MainProcess(61149):MainThread @config.py:init_config:1603 - [global]: Value for "print_" not set, using default: False
2021-07-14 17:22:33,960 [virtwho.rhsm_log DEBUG] MainProcess(61149):MainThread @config.py:init_config:1603 - [global]: Value for "log_per_config" not set, using default: False
2021-07-14 17:22:33,960 [virtwho.rhsm_log DEBUG] MainProcess(61149):MainThread @config.py:init_config:1603 - [global]: Value for "reporter_id" not set, using default: thinkpad-t580-47fed3967f99457c82472ebd22fcca15
2021-07-14 17:22:33,960 [virtwho.rhsm_log DEBUG] MainProcess(61149):MainThread @config.py:init_config:1603 - [global]: Value for "interval" not set, using default: 3600
2021-07-14 17:22:33,960 [virtwho.rhsm_log DEBUG] MainProcess(61149):MainThread @config.py:init_config:1603 - [global]: Value for "log_file" not set, using default: rhsm.log
2021-07-14 17:22:33,960 [virtwho.rhsm_log DEBUG] MainProcess(61149):MainThread @config.py:init_config:1603 - [global]: Value for "log_dir" not set, using default: /var/log/rhsm
2021-07-14 17:22:33,960 [virtwho.rhsm_log DEBUG] MainProcess(61149):MainThread @config.py:init_config:1603 - [local_libvirt]: Value for "hypervisor_id" not set, using default: uuid
2021-07-14 17:22:33,960 [virtwho.rhsm_log WARNING] MainProcess(61149):MainThread @config.py:init_config:1603 - [local_libvirt]: Ignoring unknown configuration option "rhsm_org"
2021-07-14 17:22:33,961 [virtwho.rhsm_log INFO] MainProcess(61149):MainThread @executor.py:__init__:58 - Using config named 'local_libvirt'
2021-07-14 17:22:33,961 [virtwho.rhsm_log INFO] MainProcess(61149):MainThread @main.py:main:139 - Using configuration "local_libvirt" ("libvirt" mode)
2021-07-14 17:22:33,961 [virtwho.rhsm_log INFO] MainProcess(61149):MainThread @main.py:main:142 - Using reporter_id='thinkpad-t580-47fed3967f99457c82472ebd22fcca15'
/home/jhnidek/githup/candlepin/virt-who/virtwho/manager/subscriptionmanager/subscriptionmanager.py:141: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if key is 'ssl_port':
2021-07-14 17:22:33,972 [rhsm.https DEBUG] MainProcess(61149):MainThread @https.py:<module>:56 - Using standard libs to provide httplib and ssl
2021-07-14 17:22:33,978 [virtwho.main DEBUG] MainProcess(61149):Thread-2 @virt.py:run:483 - Thread 'local_libvirt' started
2021-07-14 17:22:33,979 [virtwho.main INFO] MainProcess(61149):Thread-2 @libvirtd.py:_connect:281 - Using libvirt url: qemu:///system?no_tty=1
2021-07-14 17:22:33,980 [virtwho.main INFO] MainProcess(61149):Thread-2 @virt.py:_send_data:1154 - Report for config "local_libvirt" gathered, placing in datastore
2021-07-14 17:22:33,981 [virtwho.main DEBUG] MainProcess(61149):Thread-2 @virt.py:run:517 - Thread 'local_libvirt' stopped after running once
2021-07-14 17:22:34,980 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:__init__:172 - Environment variable NO_PROXY= will be used
2021-07-14 17:22:34,979 [virtwho.destination_8465602809756277484 DEBUG] MainProcess(61149):Thread-3 @virt.py:run:483 - Thread 'destination_8465602809756277484' started
2021-07-14 17:22:34,980 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:__init__:271 - Connection built: host=svice port=8443 handler=/candlepin auth=basic username=admin
2021-07-14 17:22:34,979 [virtwho.destination_8465602809756277484 DEBUG] MainProcess(61149):Thread-3 @subscriptionmanager.py:_connect:148 - Authenticating with RHSM username admin
2021-07-14 17:22:34,980 [virtwho.destination_8465602809756277484 INFO] MainProcess(61149):Thread-3 @subscriptionmanager.py:_connect:159 - X-Correlation-ID: 83bbe64c6bd64914865cd6a8b273699a
2021-07-14 17:22:34,982 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:717 - Making request: GET /candlepin/status/
2021-07-14 17:22:34,998 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_update_smoothed_response_time:813 - Response time: 0.007340192794799805, Smoothed response time: 0.007340192794799805
2021-07-14 17:22:34,998 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:790 - Response: status=200, requestUuid=23b28786-ea15-4a49-a68b-bee7bd9531e3, request="GET /candlepin/status/"
2021-07-14 17:22:34,999 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:717 - Making request: GET /candlepin/status
2021-07-14 17:22:35,010 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_update_smoothed_response_time:813 - Response time: 0.007576704025268555, Smoothed response time: 0.007363843917846679
2021-07-14 17:22:35,010 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:790 - Response: status=200, requestUuid=59559d51-b9f5-4ab7-9029-63e297155a42, request="GET /candlepin/status"
2021-07-14 17:22:35,011 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_load_manager_capabilities:1028 - Server has the following capabilities: ['instance_multiplier', 'derived_product', 'vcpu', 'cert_v3', 'hypervisors_heartbeat', 'remove_by_pool_id', 'syspurpose', 'storage_band', 'cores', 'hypervisors_async', 'org_level_content_access', 'guest_limit', 'ram', 'batch_bind']
2021-07-14 17:22:35,011 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:717 - Making request: PUT /candlepin/hypervisors/admin/heartbeat?reporter_id=status_test
2021-07-14 17:22:35,082 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_update_smoothed_response_time:813 - Response time: 0.007044553756713867, Smoothed response time: 0.007331914901733398
2021-07-14 17:22:35,082 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:790 - Response: status=200, requestUuid=b934e0c2-8f0a-442a-99cb-bc0a5846070e, request="PUT /candlepin/hypervisors/admin/heartbeat?reporter_id=status_test"
2021-07-14 17:22:35,084 [virtwho.destination_8465602809756277484 DEBUG] MainProcess(61149):Thread-3 @virt.py:check_report_status:914 - Existing report state: 1

<======= virt-who waits 15 seconds here for some reason

2021-07-14 17:22:50,099 [virtwho.destination_8465602809756277484 DEBUG] MainProcess(61149):Thread-3 @subscriptionmanager.py:_connect:148 - Authenticating with RHSM username admin
2021-07-14 17:22:50,099 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:__init__:172 - Environment variable NO_PROXY= will be used
2021-07-14 17:22:50,099 [virtwho.destination_8465602809756277484 INFO] MainProcess(61149):Thread-3 @subscriptionmanager.py:_connect:159 - X-Correlation-ID: 83bbe64c6bd64914865cd6a8b273699a
2021-07-14 17:22:50,100 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:__init__:271 - Connection built: host=svice port=8443 handler=/candlepin auth=basic username=admin
2021-07-14 17:22:50,101 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:717 - Making request: GET /candlepin/status/
2021-07-14 17:22:50,113 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_update_smoothed_response_time:813 - Response time: 0.007567167282104492, Smoothed response time: 0.007567167282104492
2021-07-14 17:22:50,113 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:790 - Response: status=200, requestUuid=112367ca-5ef3-4c45-a227-5eeb29c02b4e, request="GET /candlepin/status/"
2021-07-14 17:22:50,114 [virtwho.destination_8465602809756277484 DEBUG] MainProcess(61149):Thread-3 @subscriptionmanager.py:check_report_state:340 - Checking status of job ff8080817aa578d7017aa57fb5eb0bdd
2021-07-14 17:22:50,114 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:717 - Making request: GET /candlepin/jobs/ff8080817aa578d7017aa57fb5eb0bdd?result_data=True
2021-07-14 17:22:50,129 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_update_smoothed_response_time:813 - Response time: 0.008419513702392578, Smoothed response time: 0.007652401924133301
2021-07-14 17:22:50,130 [rhsm.connection DEBUG] MainProcess(61149):Thread-3 @connection.py:_request:790 - Response: status=200, requestUuid=ba0c0e1c-6701-459c-8a15-ec7284a173b6, request="GET /candlepin/jobs/ff8080817aa578d7017aa57fb5eb0bdd?result_data=True"
2021-07-14 17:22:50,130 [virtwho.destination_8465602809756277484 DEBUG] MainProcess(61149):Thread-3 @subscriptionmanager.py:check_report_state:376 - Number of mappings unchanged: 1
2021-07-14 17:22:50,131 [virtwho.destination_8465602809756277484 INFO] MainProcess(61149):Thread-3 @subscriptionmanager.py:check_report_state:377 - Mapping for config "local_libvirt" updated
2021-07-14 17:22:50,131 [virtwho.destination_8465602809756277484 DEBUG] MainProcess(61149):Thread-3 @virt.py:run:503 - Thread 'destination_8465602809756277484' terminated
{
    "configurations": [
        {
            "name": "local_libvirt",
            "source": {
                "connection": "qemu:///system?no_tty=1",
                "status": "success",
                "last_successful_retrieve": "2021-07-14 14:50:47 UTC",
                "hypervisors": 1,
                "guests": 7
            },
            "destination": {
                "connection": "svice",
                "status": "success",
                "last_successful_send": "2021-07-14 14:50:48 UTC",
                "last_successful_send_job_status": "FINISHED"
            }
        }
    ]
}
2021-07-14 17:22:50,996 [virtwho.main DEBUG] MainProcess(61149):MainThread @__main__.py:main:25 - virt-who terminated
2021-07-14 17:22:50,996 [virtwho.main DEBUG] MainProcess(61149):MainThread @main.py:exit:262 - None
2021-07-14 17:22:50,996 [virtwho.main DEBUG] MainProcess(61149):MainThread @executor.py:terminate:278 - virt-who is shutting down

real	0m17,278s
user	0m0,263s
sys	0m0,030s
```

File: tests/test_subscriptionmanager.py
Comment (wottop) [reply tests/test_subscriptionmanager.py]: Cut and pasted from another test. I changed it to your suggestion as the original was unnecessary.

**Comment by wottop** [comment]
The pause for 15 seconds was there because of the time between running the hypervisor report and checking the job status.
As the status run is not sending a report I have removed the pause prior to checking the job status. That 15 seconds should not show up whether the service is running or not.

**Review by jirihnidek — APPROVED** [review]
ACK

**Pull request merged**
